Code Challenge

DwayneCoussement opened the pull request:
# Setup Instructions
## Add a .env File:
In the root directory of the project, create a new .env file.

## Define the API_KEY:
Inside the .env file, add the following line:
> API_KEY=your_yelp_api_key_here

Replace your_yelp_api_key_here with your actual Yelp API key.

## Ensure the Project Uses the API Key:
The app will automatically load the .env file during runtime, and the API key will be used for making requests to the production environment.

jamesshaw-sf commented on lib/providers/restaurantsprovider.dart:
```diff
@@ -0,0 +1,90 @@
+import 'package:restaurantour/models/restaurant.dart';
+
+/// This is a Provider clss which extends ChangeNotifies to mange the Sate for the Restaurants.
+class RestaurantsProvider extends ChangeNotifier {
```
suggestion: file should have an underscore between words.

jamesshaw-sf commented on lib/providers/restaurantsprovider.dart:
```diff
@@ -0,0 +1,90 @@
+  /// If the given listener is not registered, the call is ignored.
+  /// This method must not be called after [dispose] has been called.
+  ///
+
```
suggestion: there shouldn't be a new line here.

jamesshaw-sf commented on lib/widgets/restaurantlistitem.dart:
```diff
@@ -0,0 +1,98 @@
+              child: Container(
+                  height: 88,
+                  margin: const EdgeInsets.only(left: 5.0, right: 15.0, top: 10.0, bottom: 10.0),
+                  child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
```
bug: overflow.
![Screen Shot 2022-07-19 at 9 10 44 AM](https://user-images.githubusercontent.com/85496350/179758387-903feaee-5f22-479e-95d1-55421bccad1a.png)

jamesshaw-sf commented on lib/widgets/restaurantdetails.dart:
```diff
@@ -0,0 +1,157 @@
+                      title: Text(restDetails.name),
+                      actions: <Widget>[
+                        Padding(
+                            padding: EdgeInsets.only(right: 20.0),
```
chore: could be const.

jamesshaw-sf commented on lib/widgets/allrestaurants.dart:
```diff
@@ -0,0 +1,69 @@
+/// Copyright 2022 - Superformula. All rights reserved.
```
chore: The designs call for more padding on this screen.

jamesshaw-sf commented on lib/widgets/restaurantdetails.dart:
```diff
@@ -0,0 +1,157 @@
+/// A widget that does not require mutable state.
+/// Show all details of the selected Restaurant by calling the Provider
+/// This Class is route from the All Restaurants/Favourite Restaurants
+///
```
chore: remove the extra ///

jamesshaw-sf commented on lib/widgets/restaurantdetails.dart:
```diff
@@ -0,0 +1,157 @@
+                                ])),
+                            const Divider(color: Colors.black),
+                            Container(alignment: Alignment.centerLeft, height: 50, child: Text("${restDetails.reviews.length} Reviews")),
+                            Container(
```
chore: this container isn't needed.

jamesshaw-sf commented on lib/widgets/restaurantdetails.dart:
```diff
@@ -0,0 +1,157 @@
+                                  Container(
+                                    alignment: Alignment.centerLeft,
+                                    height: 45,
+                                    child: const Text("Review not posted"),
```
bug:  I don't see how the actual review text would ever be displayed.

jamesshaw-sf commented on lib/widgets/restaurantdetails.dart:
```diff
@@ -0,0 +1,157 @@
+                            ),
+                            const Divider(color: Colors.black),
+                            Container(alignment: Alignment.centerLeft, height: 50, child: const Text("Address")),
+                            Container(alignment: Alignment.centerLeft, height: 50, child: Text(restDetails.location.formattedAddress ?? "")),
```
bug: the address is not displayed.
![Screen Shot 2022-07-19 at 9 23 44 AM](https://user-images.githubusercontent.com/85496350/179761044-45bf5c6b-0f4c-4e21-864d-e138245c3513.png)

jamesshaw-sf commented on lib/main.dart:
```diff
@@ -1,10 +1,98 @@
+///Start method - which is the main
+/// This is going to call the StatelessWidget
+/// runApp is the driver of the App
+void main() => runApp( RestaurantHome());
```
chore: remove extra space.

jamesshaw-sf reviewed (commented):
Thanks for taking the time to submit your code for review!  I have left a few comments.

venkatesh-seelam commented:
Hi James,

   I have fixed all issues except Reviews and Address.
   Address always returns null and Review class and User subclass doesn't
have any property for the review text.

  Code has been merged in Git and sent a pull request.

Thanks
Venkatesh

On Tue, Jul 19, 2022 at 9:27 AM jamesshaw-sf ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> Thanks for taking the time to submit your code for review! I have left a
> few comments.
> ------------------------------
>
> In lib/providers/restaurantsprovider.dart
> <https://github.com/Superformula/flutter_test/pull/7#discussion_r924478110>
> :
>
> > @@ -0,0 +1,90 @@
> +/// Copyright 2022 - Superformula. All rights reserved.
> +import 'package:flutter/material.dart';
> +import 'package:restaurantour/repositories/yelp_repository.dart';
> +import 'package:restaurantour/models/restaurant.dart';
> +
> +/// This is a Provider clss which extends ChangeNotifies to mange the Sate for the Restaurants.
> +class RestaurantsProvider extends ChangeNotifier {
>
> suggestion: file should have an underscore between words.
> ------------------------------
>
> In lib/providers/restaurantsprovider.dart
> <https://github.com/Superformula/flutter_test/pull/7#discussion_r924479113>
> :
>
> > +  List _allRestaurantsList = [];
> +
> +  Map<String?, Restaurant> _allRestaurants = {};
> +
> +  Map<String?, bool> _restMap = {};
> +
> +  Map get allRestaurantsMap => _allRestaurants;
> +  Map get allFavouriteRestaurants => _restMap;
> +  bool get isFirstLoading => _firstLoading;
> +  int get allRestaurantsListCount => _allRestaurantsList.length;
> +
> +  /// Load all Restaurants from the given Yelp API
> +  /// If the given listener is not registered, the call is ignored.
> +  /// This method must not be called after [dispose] has been called.
> +  ///
> +
>
> suggestion: there shouldn't be a new line here.
> ------------------------------
>
> In lib/widgets/restaurantlistitem.dart
> <https://github.com/Superformula/flutter_test/pull/7#discussion_r924483911>
> :
>
> > +        ),
> +        child: Row(mainAxisAlignment: MainAxisAlignment.start, children: [
> +          Container(
> +            height: 88,
> +            width: 88,
> +            margin: const EdgeInsets.only(left: 5.0, right: 15.0, top: 10.0, bottom: 10.0),
> +            decoration: BoxDecoration(
> +              borderRadius: const BorderRadius.all(Radius.circular(8)),
> +              image: DecorationImage(image: NetworkImage(image), fit: BoxFit.fill),
> +            ),
> +          ),
> +          Expanded(
> +              child: Container(
> +                  height: 88,
> +                  margin: const EdgeInsets.only(left: 5.0, right: 15.0, top: 10.0, bottom: 10.0),
> +                  child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
>
> bug: overflow.
> [image: Screen Shot 2022-07-19 at 9 10 44 AM]
> <https://user-images.githubusercontent.com/85496350/179758387-903feaee-5f22-479e-95d1-55421bccad1a.png>
> ------------------------------
>
> In lib/widgets/restaurantdetails.dart
> <https://github.com/Superformula/flutter_test/pull/7#discussion_r924488739>
> :
>
> > +      child: Column(children: [
> +        Flexible(
> +            child: resModel.isFirstLoading
> +                ? const Center(
> +                    child: SizedBox(
> +                        width: 50,
> +                        height: 50, //color: Colors.lightGreen,
> +                        child: CircularProgressIndicator(valueColor: AlwaysStoppedAnimation<Color>(Colors.black))),
> +                  )
> +                : Scaffold(
> +                    appBar: AppBar(
> +                      elevation: 0,
> +                      title: Text(restDetails.name),
> +                      actions: <Widget>[
> +                        Padding(
> +                            padding: EdgeInsets.only(right: 20.0),
>
> chore: could be const.
> ------------------------------
>
> In lib/widgets/allrestaurants.dart
> <https://github.com/Superformula/flutter_test/pull/7#discussion_r924490497>
> :
>
> > @@ -0,0 +1,69 @@
> +/// Copyright 2022 - Superformula. All rights reserved.
>
> chore: The designs call for more padding on this screen.
> ------------------------------
>
> In lib/widgets/restaurantdetails.dart
> <https://github.com/Superformula/flutter_test/pull/7#discussion_r924490948>
> :
>
> > @@ -0,0 +1,157 @@
> +/// Copyright 2022 - Superformula. All rights reserved.
> +import "package:flutter/material.dart";
> +import 'package:provider/provider.dart';
> +import 'package:restaurantour/providers/restaurantsprovider.dart';
> +
> +/// A widget that displays Restaurant details in a SingleChildScrollView.
> +/// A widget that does not require mutable state.
> +/// Show all details of the selected Restaurant by calling the Provider
> +/// This Class is route from the All Restaurants/Favourite Restaurants
> +///
>
> chore: remove the extra ///
> ------------------------------
>
> In lib/widgets/restaurantdetails.dart
> <https://github.com/Superformula/flutter_test/pull/7#discussion_r924491563>
> :
>
> > +                                height: 50,
> +                                child: Row(children: [
> +                                  Text(
> +                                    restDetails.rating.toString(),
> +                                    style: const TextStyle(
> +                                      fontWeight: FontWeight.bold,
> +                                      fontSize: 25,
> +                                      color: Colors.black,
> +                                    ),
> +                                  ),
> +                                  Container(
> +                                      height: 25, alignment: Alignment.bottomLeft, child: const Icon(Icons.star, color: Colors.orange, size: 12))
> +                                ])),
> +                            const Divider(color: Colors.black),
> +                            Container(alignment: Alignment.centerLeft, height: 50, child: Text("${restDetails.reviews.length} Reviews")),
> +                            Container(
>
> chore: this container isn't needed.
> ------------------------------
>
> In lib/widgets/restaurantdetails.dart
> <https://github.com/Superformula/flutter_test/pull/7#discussion_r924496898>
> :
>
> > +                                ])),
> +                            const Divider(color: Colors.black),
> +                            Container(alignment: Alignment.centerLeft, height: 50, child: Text("${restDetails.reviews.length} Reviews")),
> +                            Container(
> +                              child: Column(
> +                                  children: List.generate(restDetails.reviews.length, (index) {
> +                                var review = restDetails.reviews[index];
> +                                return Column(children: [
> +                                  Row(
> +                                      children: List.generate(review.rating.round(), (rindex) {
> +                                    return const Icon(Icons.star, color: Colors.orange, size: 12);
> +                                  })),
> +                                  Container(
> +                                    alignment: Alignment.centerLeft,
> +                                    height: 45,
> +                                    child: const Text("Review not posted"),
>
> bug: I don't see how the actual review text would ever be displayed.
> ------------------------------
>
> In lib/widgets/restaurantdetails.dart
> <https://github.com/Superformula/flutter_test/pull/7#discussion_r924497825>
> :
>
> > +                                      color: Colors.black,
> +                                    ),
> +                                  ),
> +                                  restDetails.isOpen
> +                                      ? const Icon(
> +                                          Icons.circle_rounded,
> +                                          size: 10,
> +                                          color: Colors.green,
> +                                        )
> +                                      : const Icon(Icons.circle_rounded, size: 10, color: Colors.red)
> +                                ])
> +                              ]),
> +                            ),
> +                            const Divider(color: Colors.black),
> +                            Container(alignment: Alignment.centerLeft, height: 50, child: const Text("Address")),
> +                            Container(alignment: Alignment.centerLeft, height: 50, child: Text(restDetails.location.formattedAddress ?? "")),
>
> bug: the address is not displayed.
> [image: Screen Shot 2022-07-19 at 9 23 44 AM]
> <https://user-images.githubusercontent.com/85496350/179761044-45bf5c6b-0f4c-4e21-864d-e138245c3513.png>
> ------------------------------
>
> In lib/main.dart
> <https://github.com/Superformula/flutter_test/pull/7#discussion_r924499247>
> :
>
> >  import 'package:flutter/material.dart';
> -import 'package:restaurantour/repositories/yelp_repository.dart';
> +import 'package:provider/provider.dart';
> +
> +import 'package:restaurantour/utils/properties.dart';
> +import 'package:restaurantour/providers/restaurantsprovider.dart';
> +import 'package:restaurantour/widgets/allrestaurants.dart';
> +import 'package:restaurantour/widgets/favouriterestaurants.dart';
> +import 'package:restaurantour/widgets/restaurantdetails.dart';
> +
> +///Start method - which is the main
> +/// This is going to call the StatelessWidget
> +/// runApp is the driver of the App
> +void main() => runApp( RestaurantHome());
>
> chore: remove extra space.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Superformula/flutter_test/pull/7#pullrequestreview-1043400711>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASH5T4ZIADYZYHRMD27F6PDVU2UN5ANCNFSM53OYMYKQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

DwayneCoussement committed:
feat: Implement List/Detail with Split/Mobile Support

- Added YelpListView to display a list of restaurants with pull-to-refresh functionality
- Integrated YelpDetailView for detailed restaurant info, with proper navigation and data passing
- Supported split-view mode for larger screens and mobile push navigation for smaller screens
- Implemented data fetching and JSON parsing using json_serializable
- Fixed layout issues and ensured smooth transition between list and detail views

DwayneCoussement committed:
chore: Add unit tests for YelpListViewModel and YelpDetailViewModel

DwayneCoussement committed:
feat: Add favorite functionality

DwayneCoussement committed:
chore: align design

DwayneCoussement committed:
Add architecture document
